### PR TITLE
[ᚬrylai-testbed] refactor: limit max peers per block download

### PR DESCRIPTION
**Pull request by zhangsoledad**

### CODE REVIEW

File: rpc/src/module/stats.rs [hunk 1]
```diff
@@ -49,15 +49,14 @@ impl<CS: ChainStore + 'static> StatsRpc for StatsRpcImpl<CS> {
     }
 
     fn get_peers_state(&self) -> Result<Vec<PeerState>> {
+        // deprecated
```
Comment (doitian) [inline rpc/src/module/stats.rs]: The deprecated info can be added to the RPC doc (in rpc/json/rpc.json and then `make gen-doc`).

File: sync/src/synchronizer/block_fetcher.rs [hunk 2]
```diff
@@ -40,23 +36,11 @@ where
...
-
-        if inflight.timestamp < unix_time_as_millis().saturating_sub(BLOCK_DOWNLOAD_TIMEOUT) {
-            debug!(
-                target: "sync", "[block downloader] inflight block download inflight.timestamp {} now {}",
-                inflight.timestamp,
-                unix_time_as_millis()
-            );
-            inflight.clear();
-        }
+    pub fn check_inflight(&self) -> bool {
```
Comment (doitian) [inline sync/src/synchronizer/block_fetcher.rs]: The name is bad. I was really surprised when I first saw the function name and it returns true when the peer reaches the limit. Let's make it more verbose, such as `reached_inflight_limit`.

File: sync/src/types.rs [hunk 3]
```diff
@@ -209,52 +212,147 @@ impl KnownFilter {
...
+    }
+}
+
+impl InflightBlocks {
+    pub fn blocks_iter(&self) -> impl Iterator<Item = (&PeerIndex, &FnvHashSet<H256>)> {
+        self.blocks.iter()
+    }
+
+    pub fn total_inflight_count(&self) -> usize {
+        self.states.len()
```
Comment (keroro520) [inline sync/src/types.rs]: Seems the behavior is ambiguity with function name?

File: sync/src/types.rs [hunk 4]
```diff
@@ -209,52 +212,147 @@ impl KnownFilter {
...
-        self.update_timestamp();
+    pub fn remove_by_block(&mut self, block: &H256) -> bool {
+        self.states
+            .remove(block)
+            .map(|state| {
+                for peer in state.peers {
+                    self.blocks.get_mut(&peer).map(|set| set.remove(block));
+                }
+            })
+            .is_some()
```
Comment (keroro520) [inline sync/src/types.rs]: Update timestamp when received a new block

**Comment by doitian** [comment]
Merged locally

**Pull request closed without merging**